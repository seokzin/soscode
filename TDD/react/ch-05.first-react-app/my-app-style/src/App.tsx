import React from 'react';
import logo from './logo.svg';
import Styled, {keyframes} from 'styled-components';

const Container = Styled.div`
  text-align: center;
`;

const Header = Styled.header`
  background-color: #282c34 !important;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const spin = keyframes`
  from {
      transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;


const AppLogo = Styled.img`
  height: 40vmin;
  pointer-events: none;

  @media (prefers-reduced-motion: no-preference) {
    animation: ${spin} infinite 20s linear;
  }
`;

const AppLink = Styled.a`
  color: #61dafb;
`;

function App() {
  return (
    // <div className="App">
    <Container>
      {/* <header className="App-header"> */}
      <Header>
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        <AppLogo src={logo} alt="logo" />
        <p>
          Edit <code>src/App.tsx</code> and save to reload.
        </p>
        <AppLink
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </AppLink>
      </Header>
      {/* </header> */}
    </Container>
    // </div>
  );
}

export default App;
