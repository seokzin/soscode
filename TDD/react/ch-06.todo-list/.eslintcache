[{"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx":"1","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx":"3","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx":"4","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx":"5","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx":"6","/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx":"7"},{"size":501,"mtime":1608450377531,"results":"8","hashOfConfig":"9"},{"size":425,"mtime":1607145602318,"results":"10","hashOfConfig":"9"},{"size":1672,"mtime":1615884272454,"results":"11","hashOfConfig":"9"},{"size":79,"mtime":1608450376614,"results":"12","hashOfConfig":"9"},{"size":989,"mtime":1608450392527,"results":"13","hashOfConfig":"9"},{"size":644,"mtime":1610068351702,"results":"14","hashOfConfig":"9"},{"size":655,"mtime":1608450377481,"results":"15","hashOfConfig":"9"},{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"1o6yy7c",{"filePath":"19","messages":"20","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"21","messages":"22","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"23","messages":"24","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"25","messages":"26","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},{"filePath":"27","messages":"28","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"29"},{"filePath":"30","messages":"31","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0,"usedDeprecatedRules":"18"},"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/index.tsx",[],["32","33"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/App.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/Input/index.tsx",[],["34","35"],"/Users/jeonghean/projects/react-book/code/ch6.todo-list/src/Components/ToDoItem/index.tsx",[],{"ruleId":"36","replacedBy":"37"},{"ruleId":"38","replacedBy":"39"},{"ruleId":"36","replacedBy":"40"},{"ruleId":"38","replacedBy":"41"},"no-native-reassign",["42"],"no-negated-in-lhs",["43"],["42"],["43"],"no-global-assign","no-unsafe-negation"]