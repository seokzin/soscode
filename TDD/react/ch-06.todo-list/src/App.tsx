import React, { useState } from 'react';
import Styled from 'styled-components';

import { Button, Input, ToDoItem } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;
const ToDoListContainer = Styled.div`
  min-width: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
`;
const InputContainer = Styled.div`
  display: flex;
`;

function App() {
  const [toDo, setToDo] = useState('');
  const [toDoList, setToDoList] = useState<string[]>([]);

  const addToDo = (): void => {
    if (toDo) {
      setToDoList([...toDoList, toDo]);
      setToDo('');
    }
  };

  const deleteToDo = (index: number): void => {
    let list = [...toDoList];
    list.splice(index, 1);
    setToDoList(list);
  };

  return (
    <Container>
      <Contents>
        <ToDoListContainer data-testid='toDoList'>
          {toDoList.map((item, index) => (
            <ToDoItem key={item} label={item} onDelete={() => deleteToDo(index)} />
          ))}
        </ToDoListContainer>
        <InputContainer>
          <Input
            placeholder="할 일을 입력해 주세요"
            value={toDo}
            onChange={(text) => setToDo(text)}
          />
          <Button label="추가" onClick={addToDo} />
        </InputContainer>
      </Contents>
    </Container>
  );
}

export default App;
