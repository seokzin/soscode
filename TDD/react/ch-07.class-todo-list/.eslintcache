[{"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Button/index.tsx":"1","/Users/jeonghean/projects/react-book/code/class-todo-list/src/reportWebVitals.ts":"2","/Users/jeonghean/projects/react-book/code/class-todo-list/src/index.tsx":"3","/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Input/index.tsx":"4","/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/ToDoItem/index.tsx":"5","/Users/jeonghean/projects/react-book/code/class-todo-list/src/App.tsx":"6"},{"size":1057,"mtime":1608810904690,"results":"7","hashOfConfig":"8"},{"size":425,"mtime":1607145602318,"results":"9","hashOfConfig":"8"},{"size":501,"mtime":1608450377531,"results":"10","hashOfConfig":"8"},{"size":706,"mtime":1608810377314,"results":"11","hashOfConfig":"8"},{"size":731,"mtime":1608810706026,"results":"12","hashOfConfig":"8"},{"size":2961,"mtime":1608882345348,"results":"13","hashOfConfig":"8"},{"filePath":"14","messages":"15","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1v95oxn",{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"18","messages":"19","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"20","messages":"21","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"22","messages":"23","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"24","messages":"25","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Button/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/reportWebVitals.ts",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/Input/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/Components/ToDoItem/index.tsx",[],"/Users/jeonghean/projects/react-book/code/class-todo-list/src/App.tsx",[]]