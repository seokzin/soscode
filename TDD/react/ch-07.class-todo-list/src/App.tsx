import React, { Component } from 'react';
import type { IScriptSnapshot } from 'typescript';
import Styled from 'styled-components';

import { Button, Input, ToDoItem } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;
const ToDoListContainer = Styled.div`
  min-width: 350px;
  height: 400px;
  overflow-y: scroll;
  border: 1px solid #BDBDBD;
  margin-bottom: 20px;
`;
const InputContainer = Styled.div`
  display: flex;
`;

interface Props {}

interface State {
  readonly toDo: string;
  readonly toDoList: string[];
  readonly error: boolean;
}

class App extends Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      toDo: '',
      toDoList: [],
      error: false,
    };
  }

  private addToDo = (): void => {
    const { toDo, toDoList } = this.state;
    if (toDo) {
      this.setState({
        toDo: '',
        toDoList: [...toDoList, toDo],
      });
    }
  };

  private deleteToDo = (index: number): void => {
    let list = [...this.state.toDoList];
    list.splice(index, 1);
    this.setState({
      toDoList: list,
    });
  };

  render() {
    const { toDo, toDoList, error } = this.state;

    return (
      <Container>
        {!error && (
          <Contents>
            <ToDoListContainer data-testid="toDoList">
              {toDoList.map((item, index) => (
                <ToDoItem key={item} label={item} onDelete={() => this.deleteToDo(index)} />
              ))}
            </ToDoListContainer>
            <InputContainer>
              <Input
                placeholder="할 일을 입력해 주세요"
                value={toDo}
                onChange={(text) => this.setState({ toDo: text })}
              />
              <Button label="추가" onClick={this.addToDo} />
            </InputContainer>
          </Contents>
        )}
      </Container>
    );
  }

  // static getDerivedStateFromProps(nextProps: Props, prevState: State) {
  //   console.log('getDerivedStateFromProps');

  //   return null;
  // }

  // componentDidMount() {
  //   console.log('componentDidMount');
  // }

  // getSnapshotBeforeUpdate(prevProps: Props, prevState: State) {
  //   console.log('getSnapshotBeforeUpdate');

  //   return {
  //     testData: true,
  //   };
  // }

  // componentDidUpdate(prevProps: Props, prevState: State, snapshot: IScriptSnapshot) {
  //   console.log('componentDidUpdate');
  // }

  // shouldComponentUpdate(nextProps: Props, nextState: State) {
  //   console.log('shouldComponentUpdate');
  //   return true;
  // }

  // componentWillUnmount() {
  //   console.log('componentWillUnmount');
  // }

  // componentDidCatch(error: Error, info: React.ErrorInfo) {
  //   this.setState({
  //     error: true,
  //   });
  // }
}
export default App;
