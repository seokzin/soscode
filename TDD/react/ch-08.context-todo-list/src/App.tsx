import React from 'react';
import Styled from 'styled-components';

import { ToDoListProvider } from 'Contexts';
import { InputContainer, ToDoList } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;
const Contents = Styled.div`
  display: flex;
  background-color: #FFFFFF;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
`;

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <Contents>
          <ToDoList />
          <InputContainer />
        </Contents>
      </Container>
    </ToDoListProvider>
  );
}

export default App;
