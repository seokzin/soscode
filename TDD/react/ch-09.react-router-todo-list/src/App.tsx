import React from 'react';
import Styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import { ToDoListProvider } from 'Contexts';
import { List, Add, Detail, NotFound } from 'Pages';
import { PageHeader } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

function App() {
  return (
    <ToDoListProvider>
      <Container>
        <PageHeader />
        <Switch>
          <Route exact path="/">
            <List />
          </Route>
          <Route path="/add">
            <Add />
          </Route>
          <Route path="/detail/:id">
            <Detail />
          </Route>
          <Route>
            <NotFound />
          </Route>
        </Switch>
      </Container>
    </ToDoListProvider>
  );
}

export default App;
