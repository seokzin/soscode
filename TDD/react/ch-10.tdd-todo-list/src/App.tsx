import React from 'react';
import Styled from 'styled-components';
import { Switch, Route } from 'react-router-dom';

import { PageHeader } from 'Components';
import { List, Add, Detail, NotFound } from 'Pages';

const Container = Styled.div`
  min-height: 100vh;
  background-color: #EEEEEE;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

function App() {
  return (
    <Container>
      <PageHeader />
      <Switch>
        <Route exact path="/">
          <List />
        </Route>
        <Route path="/add">
          <Add />
        </Route>
        <Route path="/detail/:id">
          <Detail />
        </Route>
        <Route>
          <NotFound />
        </Route>
      </Switch>
    </Container>
  );
}

export default App;
