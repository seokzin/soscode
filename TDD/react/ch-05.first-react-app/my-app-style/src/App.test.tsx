import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('renders component correctly', () => {
    const { container } = render(<App />);

    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();

    const appLogo = screen.getByAltText('logo');
    expect(appLogo).toBeInTheDocument();
    expect(appLogo).toHaveAttribute('src', 'logo.svg');

    expect(container.getElementsByTagName('p')).toHaveLength(1);
    expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
      'Edit src/App.tsx and save to reload.'
    );

    expect(container).toMatchSnapshot();
  });
});
