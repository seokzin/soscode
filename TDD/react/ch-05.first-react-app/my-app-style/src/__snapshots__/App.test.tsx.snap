// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ hMdVtH"
  >
    <div
      class="sc-gsTCUz iOxorS"
    >
      <img
        alt="logo"
        class="sc-dlfnbm eNdJmz"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="sc-hKgILt eJlGDo"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </div>
  </div>
</div>
`;
